sunpy/net/tests/test_vso.py

```diff
@@ -448,6 +448,16 @@ def test_vso_hmi(client, tmpdir):
         assert all([s == series[0] for s in series])
 
 
+@pytest.mark.remote_data
+def test_vso_mdi(client):
+    timerange = a.Time('2005/01/01 00:12', '2005/01/01 00:21')
+    result = client.search(timerange, a.Instrument('mdi'),
+                           a.Wavelength(6768*u.angstrom) & a.Physobs('LOS_magnetic_field'))
+    assert len(result) == 3
```

Comment:
What is this test actually checking? The number of DRIs in the fetch request? Can we actually the thing we want to make sure is happening rather than relying on the return values from VSO?

Reply:
I added this check which will fail on upstream but not after the fix. Though it is not checking anything; just that fix works for a query which won't work on master. So I think I should check in the test that records in getdatarequest have one fileid per record in this test.